keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1115 @@
+users specifies their required guaranteed bandwidth. Otherwise, the default value defined by the DRA driver is applied.
+
+### Risks and Mitigations
+
+- The requested amount in the resource claim may not satisfy the capacity sharing policy,
+  especially if the requested amount exceeds the maximum allowed consumption.
+
+  This scenario should be handled similarly to other scheduling issues, such as when the request exceeds the allocatable capacity. 
+  In such cases, the allocation fails and the pod remains pending.
+
+- The driver includes both sharable and unsharable devices. There is a risk that a user may be allocated a sharable device
+  (e.g., a sharable network device) and accidentally configure it with the HostDevice CNI plugin.
+  This would move the device from the host into the user's pod, preventing other users from accessing the sharable device.
+
+  **Mitigation:**
+  - Administrators should define clear device classes for sharable and unsharable devices to prevent such misallocations.
+
+## Design Details
+
+This enhancement introduces a `sharable` field within the `Device` of the ResourceSlice
+to mark whether the device is a sharable device.
+The sharable device can be assigned to more than one request if it satisfies the selection criteria and constraints.
+The select condition `device.sharable == true/false` is used to identify to select the device with a `sharable` property or not.
+
+The enhancement also adds a `SharingPolicy` field to `DeviceCapacity`.
+This field specifies how the capacity can be sharable between different requests.
+The sharing policy can be either specify a range of valid values or a discrete set of them.
+Each policy has a default value, either explicitly or implicitly.
+
+Users can define specific per-device resource requests through the newly added `CapacityRequests` field in the `DeviceRequest`. Each contains a minimum required device capacity.
+If the capacity is consumable, the amount available for allocation is determined
+by subtracting the aggregated allocation results of current claims from the device's capacity as defined in the resource slice.
+The remaining amount will be used solely by the allocator and will not be reflected in the resource slice.
+The calculation of capacity requirements will round the requested capacity up to the nearest valid amount,
+based on the capacity's sharing policy.
+
+If users do not specify a capacity request for consumable capacity, the default consumed value will be applied.
+There is always such a default because capacities without a policy are not consumable.
+
+A sharable device can only be allocated once its consumability has been verified
+and its attributes match the request's selectors and constraints.
+The newly added `capacities` field in the `DeviceRequestAllocationResult` will be set when the allocation is successful.
+
+### API enhancement
+To enable this enhancement, the following API updates are proposed.
```

Comment:
Adding `+listMapKey=shareUID` to

https://github.com/kubernetes/kubernetes/blob/5ec7b65318afb91b8d77f3f262ee0b93dcd84b5f/staging/src/k8s.io/api/resource/v1beta2/types.go#L1144-L1150

needs to be mentioned.